src/main/java/net/sf/jabref/logic/importer/fetcher/MrDLibFetcher.java

```diff
@@ -0,0 +1,145 @@
+        MrDLibImporter importer = new MrDLibImporter();
+        ParserResult parserResult = new ParserResult();
+        try {
+            if (importer.isRecognizedFormat(new BufferedReader(new StringReader(response)))) {
+                parserResult = importer.importDatabase(new BufferedReader(new StringReader(response)));
+            } else {
+                // For displaying An ErrorMessage
+                BibEntry errorBibEntry = new BibEntry();
+                errorBibEntry.setField("html_representation",
+                        Localization.lang("Error_while_fetching_from_%0", "Mr.DLib"));
+                BibDatabase errorBibDataBase = new BibDatabase();
+                errorBibDataBase.insertEntry(errorBibEntry);
+                parserResult = new ParserResult(errorBibDataBase);
+            }
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        return parserResult.getDatabase().getEntries();
+    }
+
+    /**
+     * Contact the server with the title of the selected item
+     * @param query: The query holds the title of the selected entry. Used to make a query to the MDL Server
+     * @return Returns the server response. This is an XML document as a String.
+     * @throws FetcherException
+     */
+    private String makeServerRequest(String query) throws FetcherException {
+        query = constructQuery(query);
+        String response = "";
+        SSLContext sslContext = null;
+
+        try {
+            sslContext = SSLContexts.custom().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build();
+        } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e1) {
+            LOGGER.error(e1.getMessage(), e1);
+        }
+
+        SSLConnectionSocketFactory sslSocketFacktory = new SSLConnectionSocketFactory(sslContext);
+
+        try (CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslSocketFacktory).build()) {
+            Unirest.setHttpClient(httpclient);
+            try {
+                response = Unirest.get(query).asString().getBody();
+            } catch (UnirestException e) {
+                LOGGER.error(e.getMessage(), e);
+                throw new FetcherException(e.getMessage(), Localization.lang("Error_while_fetching_from_%0", "Mr.DLib"),
+                        e);
+            }
+            //Conversion of < and >
+            response = response.replaceAll("&gt;", ">");
+            response = response.replaceAll("&lt;", "<");
+        } catch (IOException e1) {
+            LOGGER.error(e1.getMessage(), e1);
+        }
+        return response;
+    }
+
+    /**
+     * Constructs the query based on title of the bibentry. Adds statistical stuff to the url.
+     * @param query: the title of the bib entry.
+     * @return the string used to make the query at mdl server
+     */
+    private String constructQuery(String query) {
+        StringBuffer queryBuffer = new StringBuffer();
```

Comment:
This is just about code style. Please have a look at http://www.leveluplunch.com/java/examples/construct-build-uri/  "Construct URI w/ query parameter"